packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -270,6 +270,14 @@ export default class CallDiagnosticLatencies extends WebexPlugin {
     return this.precomputedLatencies.get('internal.client.pageJMT') || undefined;
   }
 
+  /**
+   * Download Time JMT
+   * @returns - latency
+   */
+  public getDownloadTimeJMT() {
+    return this.precomputedLatencies.get('internal.download.time') || undefined;
```

Comment:
There is no place, within this branch, that `'internal.download.time'` is being explicitly assigned to `precomputedLatencies`, so this should always return `undefined`. Is this being handled via spread/map elsewhere? If so, we should potentially introduce some form of map/document/interface for maintainability of this segment of code.

**I would recommend**: moving away from `Map` instances unless you're specifically need to **key** an `Object` with `Object` instances.

Also, is there redundancy with `/* ... */ || undefined`? Just want to confirm, as I may be misunderstanding this area.

Reply:
That's because we're setting that in Cantina and not in SDK. Some latencies are precomputed in Cantina and we just save them all here in one place.

|| undefined was added I think because in case latencies/timestamps equal to 0, would rather return undefined. CA doesn't want 0 values.

Reply:
Any reason why Map is problematic?
Also, the timestamps benefit from the insertion order of the Map. The precomputedLatencies, I guess that can be changed to {} if you really think it's worth it.

Reply:
`Map` instances are not particularly problematic (hence the non-blocking). They don't support the level of type-enforcement you can get from typical interfaces when applied to objects:

```ts
const myData: MyData = {}; // fully typed, inspectable, exportable
const map: Map<string, any> = new Map(); // similar interface and generic params as `Record<string, any>`
```

In addition to this, the benefits of a `Map` instance, when compared to a traditional `Object` are not used here. We do not key the `Map` instance using any mutable data, so it is used as if it were an `Object` with a `get()` and `set()` method, losing the ability to inspect/cast and shape the data being set as a `Map`.

Let me know if there is a benefit to using a `Map` instance here if you could, as it may change my perception of this area of code.

Reply:
added types for the precomputed map